src/index.ts

```diff
@@ -21,18 +21,21 @@ const isSSR =
 
 const useIsomorphicLayoutEffect = isSSR ? useEffect : useLayoutEffect
 
-export interface UseStore<T extends State> {
+export interface UseStore<T extends State, CustomSetState = SetState<T>> {
```

Comment:
does `CustomSetState` not need an upper bound here as well?


```suggestion
export interface UseStore<T extends State, CustomSetState extends SetState<T> = SetState<T>> {
```

Reply:
Actually, this was my misunderstanding. Let me revert it.